### PR TITLE
test: test docker container pulling

### PR DESCRIPTION
**Pull request by GregorioDiStefano**
Signed-off-by: Greg Di Stefano <greg.distefano+github@gmail.com>

### CODE REVIEW

File: tests/common_setup.py [hunk 1]
```diff
@@ -139,20 +139,21 @@ def standard_setup_with_short_lived_token():
...
-    # so that is cleaned up correctly on test failure/error
+        def fin():
+            stop_docker_compose()
+            conftest.production_setup_lock.release()
 
-    def fin():
-        conftest.production_setup_lock.release()
-        stop_docker_compose()
+        # this additional lock is required
+        conftest.production_setup_lock.acquire()
```
Comment (kacf) [inline tests/common_setup.py]: This looks wrong, you're acquiring the lock twice here, once using `with`, and once here. When do you want to hold the lock, during the whole production environment being up, or only during initialization and teardown?

**Comment by GregorioDiStefano** [comment]
Filelock python module uses a lock counter. There is a lock used when starting the backend (which is then released) and an additional lock used until the finalizer releases it.

**Comment by kacf** [comment]
Ok... But what is that supposed to achieve? AFAICS you could get the exact same behavior, with less complexity, by getting rid of the `with` and acquiring the lock immediately after entering the `running_custom_production_setup`. This would also be easier to understand when reading it.

**Comment by GregorioDiStefano** [comment]
Yes, your suggestion should work. Can you amend my PR and merge this?

**Comment by kacf** [comment]
Sure, after I fix the tenant test breakage.

**Comment by kacf** [comment]
> Yes, your suggestion should work. Can you amend my PR and merge this?

Actually, looking closer, this is exactly what the old version does, and also the correct thing to do. So no action required here. :-)

**Pull request closed without merging**